crates/rune/src/compile/v1/assemble.rs

```diff
@@ -2148,6 +2159,16 @@ fn expr_field_access(
                 let field = ident.resolve(resolve_context!(c.q))?;
                 let slot = c.q.unit.new_static_string(span, field.as_ref())?;
 
+                #[cfg(feature = "dynamic_fields")]
+                c.asm.push(
+                    if c.options.dynamic_fields {
+                        Inst::ObjectIndexDynamicGet { slot }
+                    } else {
+                        Inst::ObjectIndexGet { slot }
+                    },
+                    span,
+                );
+                #[cfg(not(feature = "dynamic_fields"))]
                 c.asm.push(Inst::ObjectIndexGet { slot }, span);
```

Comment:
Yeah easily sorted; it is just like this due to `ObjectIndexDynamicGet` originally being feature gated so it wouldn't compile with that if statement lol (kind of wish rust had a macro which mimicked if else cfg but actually ignored the invalid branch since its more sane to read.